packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -363,6 +411,104 @@ contract OPContractsManager is ISemver {
         return output;
     }
 
+    /// @notice addGameType deploys a new dispute game and links it to the DisputeGameFactory. The inputted _gameConfigs
+    /// must be added in ascending GameType order.
+    function addGameType(AddGameInput[] memory _gameConfigs) external returns (AddGameOutput[] memory) {
+        if (address(this) == address(thisOPCM)) revert OnlyDelegatecall();
+        if (_gameConfigs.length == 0) revert InvalidGameConfigs();
+
+        AddGameOutput[] memory outputs = new AddGameOutput[](_gameConfigs.length);
+        Blueprints memory bps = thisOPCM.blueprints();
+
+        // Store last game config to as an int64 so that we can ensure that the same game config is not added twice.
+        // An int64 is OK to use here since the GameType is a uint32. We start at -1 since 0 is a valid GameType.
+        int64 lastGameConfig = -1;
+
+        for (uint256 i = 0; i < _gameConfigs.length; i++) {
+            AddGameInput memory gameConfig = _gameConfigs[i];
+
+            // This conversion is safe because the GameType is a uint32, which will always fit in an int64.
```

Comment:
Comments are good up to here. Beyond this point we could use more to guide the reader through the flow.

Reply:
added a bunch of comments, lmk if you think more guidance is needed